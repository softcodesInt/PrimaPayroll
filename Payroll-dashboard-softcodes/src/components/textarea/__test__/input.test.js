/* -------------------------- External Dependecies -------------------------- */

import React from 'react';
import ReactDOM from 'react-dom';
import { render, cleanup } from '@testing-library/react';
import renderer from 'react-test-renderer';

import '@testing-library/jest-dom';
import '@testing-library/jest-dom/extend-expect';

/* --------------------------- Internal Dependency -------------------------- */
import Input from '..';

const InputMock = ({ isInvalid = false, type = 'text' }) => {
	return (
		<Input
			id="input"
			type={type}
			placeholder="Enter Text Here"
			hasStrip
			label="text"
			isInvalid={isInvalid}
			errorMessage="Fill Input Out"
			value="12345"
			onChange={(e) => e.target.value}
		/>
	);
};
describe('Input Component', () => {
	afterEach(cleanup);

	it('Input Renders Without Crashing', () => {
		const div = document.createElement('div');
		ReactDOM.render(<InputMock />, div);
		ReactDOM.unmountComponentAtNode(div);
	});

	it('Input Renders Input Correctly With Attributes', () => {
		const { getByTestId } = render(<InputMock />);
		// console.log();
		expect(getByTestId('input')).not.toHaveAttribute('disabled');
	});

	it('Input matches snapshot', () => {
		const tree = renderer.create(<InputMock />).toJSON();
		expect(tree).toMatchSnapshot();
	});
	it('Input matches snapshot when type is password', () => {
		const tree = renderer
			.create(<InputMock isInvalid={true} type="password" />)
			.toJSON();
		expect(tree).toMatchSnapshot();
	});
	it('Test click event', () => {
		const component = renderer.create(
			<Input
				id="input"
				type="password"
				placeholder="Enter Text Here"
				hasStrip
				label="text"
				isInvalid
				errorMessage="Fill Input Out"
				value=""
			/>
		);
		renderer.act(() => {
			component.root.findByType('button').props.onClick();
		});
	});
});
