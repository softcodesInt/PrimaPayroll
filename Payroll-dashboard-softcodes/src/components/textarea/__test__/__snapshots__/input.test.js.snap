// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input Component Input matches snapshot 1`] = `
<div
  className="sc-bdnylx cUrMpJ"
  data-testid="input"
>
  <label
    htmlFor="text"
  >
    text
  </label>
  <input
    className="form-control"
    data-testid="inputs"
    id="input"
    onChange={[Function]}
    placeholder="Enter Text Here"
    type="text"
    value="12345"
  />
</div>
`;

exports[`Input Component Input matches snapshot when type is password 1`] = `
<div
  className="sc-bdnylx cUrMpJ"
  data-testid="input"
>
  <label
    htmlFor="text"
  >
    text
  </label>
  <input
    className="form-control"
    data-testid="inputs"
    id="input"
    onChange={[Function]}
    placeholder="Enter Text Here"
    type="password"
    value="12345"
  />
  <button
    className="input-icon "
    onClick={[Function]}
    type="button"
  >
    <svg>
      eye.svg
    </svg>
  </button>
  <span
    className="sc-gtssRu kbMXaR error"
    color="#f66262"
    fontSize="12px"
  >
    Fill Input Out
  </span>
</div>
`;
